chain/chain/src/trie_store_resharder.rs

```diff
@@ -0,0 +1,164 @@
+//! Logic for writing trie nodes with the appropriate shard prefix.
+//! This logic runs in parallel to chain processing.
+
+use crate::types::RuntimeAdapter;
+use near_chain_configs::{MutableConfigValue, ReshardingConfig};
+use near_primitives::{hash::CryptoHash, state};
+use near_store::{ShardUId, adapter::StoreAdapter, flat::ParentSplitParameters};
+use std::sync::Arc;
+
+pub struct TrieStoreResharder {
+    runtime: Arc<dyn RuntimeAdapter>,
+    /// Configuration for resharding.
+    resharding_config: MutableConfigValue<ReshardingConfig>,
+}
+
+impl TrieStoreResharder {
+    pub fn new(
+        runtime: Arc<dyn RuntimeAdapter>,
+        resharding_config: MutableConfigValue<ReshardingConfig>,
+    ) -> Self {
+        Self { runtime, resharding_config }
+    }
+
+    fn split_shard_task_impl(
+        &self,
+        shard_uid: ShardUId,
+        state_root: CryptoHash, // TODO: fix
+        parent_split_parameters: ParentSplitParameters,
+    ) -> Result<(), String> {
```

Comment:
Can you clarify what those arguments are exactly? 
1) Is it the parent_shard_uid? If so please rename. 
2) Is it the post state root of the parent shard in the resharding block? Is it a pre state root of either of the children in the block after the resharding block? Please add comments. FYI resharding block is the last block of the old shard layout